pydm/main_window.py

```diff
@@ -370,6 +372,10 @@ def open(self, filename, macros=None, args=None, target=None):
             self.ui.actionEdit_in_Designer.setText(edit_in_text)
             if (self.designer_path and ui_file) or (py_file and not ui_file):
                 self.ui.actionEdit_in_Designer.setEnabled(True)
+
+        # Create and run the thread for establishing the channel connections which have been queued above
+        establish_connections_thread = threading.Thread(target=data_plugins.establish_queued_connections, daemon=True)
```

Comment:
A few thoughts here:

If kept as-is:
1. The thread object should _probably_ be retained in the main window instance (`self._establish_connections_thread`?)
2. You might consider a private method on main window such that you can perform things before/after the `data_plugins.establish_queued_connections()` call - such as adding logging around it, or cleanup in the case of an exception, etc